src/main/java/frc/robot/subsystems/intake/Intake.java

```diff
@@ -0,0 +1,296 @@
+      ),
+      Commands.waitSeconds(1).andThen(
+        Commands.runOnce(() -> {
+          for (int i = 0; i < 4; i++) {
+            checkMotorVelocity(intakeMotors[i]);
+          }
+        })
+      )
+    );
+  }
+
+  private void checkMotorVelocity(IntakeMotor motor) {
+    if (motor == IntakeMotor.RIGHT_ROLLER) {
+      if (Math.abs(inputs.rollerVelocityRPS 
+      - inputs.rollerReferenceVelocityRPS) 
+      > IntakeConstants.ROLLER_VELOCITY_TOLERANCE) {
+        if (Math.abs(inputs.rollerVelocityRPS) >
+        Math.abs(inputs.rollerReferenceVelocityRPS)) {
+          FaultReporter.getInstance().addFault(
+            SUBSYSTEM_NAME,
+            "[System Check] RRoller Too Fast"
+          );
+        } else {
+          FaultReporter.getInstance().addFault(
+            SUBSYSTEM_NAME,
+            "[System Check] Roller Too Slow"
+          );
+        }
+      }
+    } else if (motor == IntakeMotor.KICKER) {
+      if (inputs.kickerVelocityRPS 
+      - inputs.kickerReferenceVelocityRPS 
+      > IntakeConstants.KICKER_VELOCITY_TOLERANCE) {
+        FaultReporter.getInstance().addFault(
+          SUBSYSTEM_NAME,
+          "[System Check] Kicker Too Slow"
+        );
+      }
+    }
+  }
+
+  private Command getStatesCheckCommand() {
+    IntakeState[] desiredStateSequence = {
+      IntakeState.EMPTY,
+      IntakeState.NOTE_IN_INTAKE,
+      IntakeState.NOTE_IN_INTAKE_AND_KICKER,
+      IntakeState.NOTE_IN_KICKER,
+      IntakeState.NOTE_IN_KICKER_AND_SHOOTER,
+      IntakeState.NOTE_IN_SHOOTER
+    };
+    
+    ArrayList<IntakeState> actualStateSequence = new ArrayList<>();
+
+    return Commands.parallel(
+      Commands.parallel(
+        Commands.run(() -> this.setIntakeState(IntakeState.EMPTY)),
+        Commands.run(this::intakeGamePiece)
+      ),
+      Commands.waitSeconds(1).andThen(
+        Commands.run(() -> checkStateSequence(actualStateSequence, desiredStateSequence))
+      ).until(this::checkComplete)
+    );
+  }
+
+  private void checkStateSequence(ArrayList<IntakeState> actualStateSequence, IntakeState[] desiredStateSequence) {
+    if (intakeState != mostRecentIntakeState) {
+      actualStateSequence.add(intakeState);
+      mostRecentIntakeState = intakeState;
+    }
+
+    for (int i = 0; i < actualStateSequence.size(); i++) {
+      if (!actualStateSequence.get(i).equals(desiredStateSequence[i])) {
+        FaultReporter.getInstance().addFault(
+          SUBSYSTEM_NAME,
+          "[System Check] Intake State Incorrect, "
+          + "Expected: " + desiredStateSequence[i] + ", "
+          + "Actual: " + actualStateSequence.get(i)
+        );
+      }
+    }
+
+    if (actualStateSequence.size() == desiredStateSequence.length) {
+      completeCheck();
+    }
+
+  }
+
+  
+
+  public boolean manualOverrideEnabled() {
+    return manualOverrideEnabled;
+  }
+
+  public boolean runningManualIntake() {
+    return manualOverrideEnabled
+        && Math.abs(inputs.rollerReferenceVelocityRPS) > 0;
```

Comment:
the velocity may not be exactly 0 when when not moving. You may want > 0.01 or something small....